src/components/page-header.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from "react";
+
+import { BlockMapType, MapPageUrl } from "../types";
+import PageIcon from "./page-icon";
+
+interface PageHeaderProps {
+  blockMap: BlockMapType;
+  mapPageUrl: MapPageUrl;
+}
+
+const PageHeader: React.FC<PageHeaderProps> = ({ blockMap, mapPageUrl }) => {
+  const blockIds = Object.keys(blockMap);
+  const activePageId = blockIds[0];
+
+  if (!activePageId) {
+    return null;
+  }
+
+  const breadcrumbs = [];
+  let currentPageId = activePageId;
+
+  do {
```

Comment:
here we're aggregating the data we need to render the header nav breadcrumbs:

<img width="735" alt="breadcrumbs" src="https://user-images.githubusercontent.com/552829/85630775-48945980-b642-11ea-807b-5ff937b0c91d.png">